core/server/kustomization_test.go

```diff
@@ -97,6 +98,108 @@ func TestCreateKustomization(t *testing.T) {
 	})
 }
 
+func TestListKustomizations(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := context.Background()
+
+	c, cleanup := makeGRPCServer(k8sEnv.Rest, t)
+	defer cleanup()
+
+	_, k, err := kube.NewKubeHTTPClientWithConfig(k8sEnv.Rest, "")
+	g.Expect(err).NotTo(HaveOccurred())
+
+	appName := "myapp"
+	ns := newNamespace(ctx, k, g)
+
+	r := &pb.AddKustomizationReq{
+		Name:      "mykustomization",
+		Namespace: ns.Name,
+		AppName:   appName,
+		SourceRef: &pb.SourceRef{
+			Kind: pb.SourceRef_GitRepository,
+			Name: "othersource",
+		},
+	}
+
+	addRes, err := c.AddKustomization(ctx, r)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(addRes.Success).To(BeTrue())
+
+	unAssociatedKustomizationReq := &pb.AddKustomizationReq{
+		Name:      "otherkustomization",
+		Namespace: ns.Name,
+		AppName:   "",
+		SourceRef: &pb.SourceRef{
+			Kind: pb.SourceRef_GitRepository,
+			Name: "othersource",
+		},
+	}
+
+	_, err = c.AddKustomization(ctx, unAssociatedKustomizationReq)
+	g.Expect(err).NotTo(HaveOccurred())
+
```

Comment:
might want to add `g.Expect(addRes.Success).To(BeTrue())` here

Reply:
I deliberately left if off since that isn't what this test is trying to prove.